exp/zapslog/handler.go

```diff
@@ -157,29 +162,45 @@ func (h *Handler) Handle(ctx context.Context, record slog.Record) error {
 		ce.Stack = stacktrace.Take(3 + h.callerSkip)
 	}
 
-	fields := make([]zapcore.Field, 0, record.NumAttrs())
+	fields := make([]zapcore.Field, 0, record.NumAttrs()+1)
 	record.Attrs(func(attr slog.Attr) bool {
-		fields = append(fields, convertAttrToField(attr))
+		f := convertAttrToField(attr)
+		if h.holdGroup != "" && f != zap.Skip() {
+			fields = append(fields, zap.Namespace(h.holdGroup))
+			h.holdGroup = ""
+		}
+		fields = append(fields, f)
```

Comment:
I don't think this bit is right. We cannot modify `h`'s internal state in a `Handle`.
If we have two concurrent Handle calls, this will include the group in only one of those calls.
I think what we want to do here is record a boolean in the closure outside the Attrs loop, and flip it after the first `zap.Namespace`.

Something like this:

```
var addedNamespace bool
record.Attrs(... {
  if ... && !addedNamespace {
    fields = append(...)
    addedNamespace = true
  }
})
```

This way two concurrent calls will not fight each other on this.

Reply:
fixed.